zebra-state/src/service/finalized_state.rs

```diff
@@ -838,7 +838,8 @@ impl FinalizedState {
 // up automatically eventually.
 impl Drop for FinalizedState {
     fn drop(&mut self) {
-        self.delete_ephemeral()
+        self.db.cancel_all_background_work(false);
```

Comment:
Unfortunately, this change will not make the difference.

Reply:
That's a shame, I was hoping it would be something simple.

I'll have a look at this today, because it used to work in Zebra, and it works for other projects. So it should be something we can fix.

Reply:
You're right - this argument doesn't seem to make much of a difference on my machine. But it's a very fast machine, so I'm not sure I would see any bugs here.

Waiting is the correct thing to do, so I fixed this function call in https://github.com/oxarbitrage/zebra/pull/164